Library/Homebrew/utils/curl.rb

```diff
@@ -58,10 +58,10 @@ def curl_args(*extra_args, **options)
         args << "--silent" unless $stdout.tty?
       end
 
-      args << "--connect-timeout" << connect_timeout.round(3) if options[:connect_timeout]
-      args << "--max-time" << max_time.round(3) if options[:max_time]
+      args << "--connect-timeout" << options[:connect_timeout].round(3) if options[:connect_timeout].is_a?(Numeric)
```

Comment:
Seems a bit strange that this will simply silently discard the value of e.g. `options[:connect_timeout] = "3"`. Feels more Ruby-like to either convert it to the type we expect or `raise` if it's the wrong type. Also, while I'm being pedantic, `**options` feels A Bit Ugly now we have kwargs and Sorbet so would be better refactored to use them.

Reply:
> Seems a bit strange that this will simply silently discard the value of e.g. `options[:connect_timeout] = "3"`. Feels more Ruby-like to either convert it to the type we expect or `raise` if it's the wrong type.

I was getting that feeling, too. I'll add some logic that `raise`s on a non-numeric value in the next push, as that's an easy solution.

---

> Also, while I'm being pedantic, `**options` feels A Bit Ugly now we have kwargs and Sorbet so would be better refactored to use them.

Just to make sure I understand, are you suggesting making these options into named parameters, like so:

```ruby
sig {
  params(
    extra_args:      T::Array[T.untyped],
    connect_timeout: T.any(Integer, Float, NilClass),
    max_time:        T.any(Integer, Float, NilClass),
    retry_max_time:  T.any(Integer, Float, NilClass),
    options:         T.nilable(T::Hash[Symbol, T.untyped]),
  ).returns(T::Array[T.untyped])
}
def curl_args(*extra_args, connect_timeout: nil, max_time: nil, retry_max_time: nil, **options)
```

`retry` is a Ruby keyword, so we still have to use `options[:retry]` to allow us to use `retry: false` when calling `#curl_args`.

Also, we haven't enabled typechecking for `Utils::Curl` yet (there are ~40 issues to resolve first), so the signature won't do anything right now. Enforcing the correct type for these arguments within `#curl_args` (as mentioned above) is necessary anyway, as args/options will sometimes come from places where `brew typecheck` doesn't look (e.g., a `livecheck` block in a formula/cask).

Reply:
> Just to make sure I understand, are you suggesting making these options into named parameters, like so:

Yes!

> `retry` is a Ruby keyword, so we still have to use `options[:retry]` to allow us to use `retry: false` when calling `#curl_args`.

👍🏻 but I think it may be worth a deprecation (or rename if this a private API which I think it is?) for this reason.

Reply:
The `retry` option is only used in `Livecheck::Strategy` at the moment (I added it specifically for livecheck), so it would be easy to rename it to something that can be used as a named parameter. `#curl_args` is listed as private in the docs, so renaming sounds fine.

Do you have any ideas for an alternative name for `retry`? When we were discussing this in the related PR, alternative names felt clunky (e.g., `retry_enabled`, `should_retry`, etc.) and `retry` was technically possible with the `options` setup, so we went that route. If not, I'll think on this later and see if I can come up with anything reasonable.

Reply:
> The `retry` option is only used in `Livecheck::Strategy` at the moment (I added it specifically for livecheck), so it would be easy to rename it to something that can be used as a named parameter. `#curl_args` is listed as private in the docs, so renaming sounds fine.

👍🏻 

> should_retry

This sounds good given keyword.



> Thinking about this some more, one other way to approach this would be to replace `retry` with a `retries` parameter (defaulting to `Homebrew::EnvConfig.curl_retries`) and we would only add the `--retry` arg when `retries.positive?` is true. This would allow us to use `retries: 0` to avoid the default `--retry`.

This also works. I like the elegance here.